src/rust/src/webrtc/peer_connection_factory.rs

```diff
@@ -250,11 +250,10 @@ impl PeerConnectionFactory {
                 hide_ip,
                 ice_servers.rffi(),
                 outgoing_audio_track.rffi(),
-                if let Some(outgoing_video_track) = outgoing_video_track {
-                    outgoing_video_track.rffi()
-                } else {
-                    std::ptr::null()
-                },
+                outgoing_video_track.map_or_else(
+                    || std::ptr::null(),
```

Comment:
Can you check this?

```
$ cargo clippy
   Compiling ringrtc v2.9.2 (/ringrtc/src/rust)
warning: redundant closure found
   --> src/webrtc/peer_connection_factory.rs:254:21
    |
254 |                     || std::ptr::null(),
    |                     ^^^^^^^^^^^^^^^^^^^ help: remove closure as shown: `std::ptr::null`
    |
    = note: `#[warn(clippy::redundant_closure)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure

warning: 1 warning emitted
```

Reply:
false positive

diff:
```
diff --git a/src/rust/src/webrtc/peer_connection_factory.rs b/src/rust/src/webrtc/peer_connection_factory.rs
index 429cfb1..f2daad5 100644
--- a/src/rust/src/webrtc/peer_connection_factory.rs
+++ b/src/rust/src/webrtc/peer_connection_factory.rs
@@ -250,10 +250,9 @@ impl PeerConnectionFactory {
                 hide_ip,
                 ice_servers.rffi(),
                 outgoing_audio_track.rffi(),
-                outgoing_video_track.map_or_else(
-                    || std::ptr::null(),
-                    |outgoing_video_track| outgoing_video_track.rffi(),
-                ),
+                outgoing_video_track.map_or_else(std::ptr::null(), |outgoing_video_track| {
+                    outgoing_video_track.rffi()
+                }),
                 enable_dtls,
                 enable_rtp_data_channel,
             )
```
error:
```
error[E0277]: expected a `FnOnce<()>` closure, found `*const _`
   --> src/webrtc/peer_connection_factory.rs:253:50
    |
253 |                 outgoing_video_track.map_or_else(std::ptr::null(), |outgoing_video_track| {
    |                                                  ^^^^^^^^^^^^^^^^ expected an `FnOnce<()>` closure, found `*const _`
    |
    = help: the trait `FnOnce<()>` is not implemented for `*const _`
    = note: wrap the `*const _` in a closure with no arguments: `|| { /* code */ }`
```

Reply:
You could try without the '()'.

Reply:
True, fixed.

Reply:
It looks good, we will merge it to the next release, coming soon. Thanks!